packages/azure-http-specs/specs/azure/resource-manager/resources/location.tsp

```diff
@@ -0,0 +1,175 @@
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+using Spector;
+
+namespace Azure.ResourceManager.Resources;
+
+@resource("locationResources")
+@parentResource(ResourceGroupLocationResource)
+model LocationResource is ProxyResource<LocationResourceProperties> {
+  ...ResourceNameParameter<LocationResource>;
+}
+
+/** Location resource properties */
+model LocationResourceProperties {
+  @doc("The description of the resource.")
+  description?: string;
+
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+}
+
+@armResourceOperations
+interface LocationResources {
+  @scenario
+  @scenarioDoc("""
+    Resource GET operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Resources/locations/eastus/locationResources/resource
+    Expected query parameter: api-version=2023-12-01-preview
+    
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Resources/locations/eastus/locationResources/resource",
+      "name": "resource",
+      "type": "Azure.ResourceManager.Resources/locationResources",
+      "properties":{
+        "description": "valid",
+        "provisioningState": "Succeeded"
+      },
+      "systemData": {
+        "createdBy": "AzureSDK",
+        "createdByType": "User",
+        "createdAt": <any date>,
+        "lastModifiedBy": "AzureSDK",
+        "lastModifiedAt": <any date>,
+        "lastModifiedByType": "User",
+      }
+    }
+    ```
+    """)
+  get is ArmResourceRead<LocationResource>;
+
+  @scenario
+  @scenarioDoc("""
+    Resource PUT operation.
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Resources/locations/eastus/locationResources/resource
+    Expected query parameter: api-version=2022-12-01-preview
+    Expected request body:
+    ```json
+    {
+      "properties": {
+        "description": "valid",
+      }
+    }
+    ```
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.Resources/locations/eastus/locationResources/resource",
+      "name": "resource",
+      "type": "Azure.ResourceManager.Resources/locationResources",
+      "properties": {
+        "description": "valid",
+        "provisioningState": "Succeeded",
+      },
+      "systemData": {
+        "createdBy": "AzureSDK",
+        "createdByType": "User",
+        "createdAt": <any date>,
+        "lastModifiedBy": "AzureSDK",
+        "lastModifiedAt": <any date>,
+        "lastModifiedByType": "User",
+      }
+    }
+    ```
+    """)
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<LocationResource>;
```

Comment:
Why do we use Async? The mockapi is sync.

Reply:
The `ArmResourceCreateOrUpdateAsync`  has been changed to `ArmResourceCreateOrReplaceSync` in the [commit#275b741](https://github.com/Azure/typespec-azure/pull/1932/commits/275b7410431b32eac1dd16308c3a58bca452e4e3)